plasmapy/particles/particle_class.py

```diff
@@ -2045,6 +2049,13 @@ class CustomParticle(AbstractPhysicalParticle):
     """
 
     def __init__(self, mass: u.kg = None, charge: (u.C, Real) = None, symbol=None):
+
+        if (
+            isinstance(mass, u.Quantity)
+            and u.get_physical_type(mass) == "electrical charge"
+        ):
+            charge, mass = mass, charge
+
```

Comment:
This seems like an odd approach to me...  Why not utilize `@validate_quantities` and enforce keyword only?  That seems like a more robust option and would add unit validation that does not currently exist here.

```python
    @validate_quantities(
        mass={"none_shall_pass": True},
        charge={"none_shall_pass": True},
    )
    def __init__(self, *, mass: u.kg = None, charge: u.C = None, symbol=None):
        ...
```

Reply:
If you want to help with depreciation, then you could do something like....

```python
    @validate_quantities(
        mass={"none_shall_pass": True},
        charge={"none_shall_pass": True},
    )
    def __init__(self, *args, mass: u.kg = None, charge: u.C = None, symbol=None):

        if len(*args) != 0:
             # say you depreciation waning and take mass = args[0] and charge = args[1]
            ...
        ...
```

Reply:
Have you tried Erik's suggested approach here, Nick? 

Suppose we live in an ideal world and we have MyPy for astropy Quantities. From that perspective, if we go with the current version, from the outside it looks like you'd be forcing a `charge` block into a `mass`-shaped hole and vice versa. That ideal-world unit-type-checker would rightfully complain - and anyone reading any code using this particular feature would likely be a smidge confused!

I kind of like the first version proposed by Erik here, but obviously deprecation period etc etc, so we'd have to go with the latter for now.